core/src/services/azfile/core.rs

```diff
@@ -413,15 +412,11 @@ impl AzfileCore {
     }
 
     pub async fn ensure_parent_dir_exists(&self, path: &str) -> Result<()> {
-        let mut dirs = VecDeque::default();
         // azure file service does not support recursive directory creation
         let mut p = path;
         while p != "/" {
             p = get_parent(p);
-            dirs.push_front(p);
-        }
-        for dir in dirs {
-            let resp = self.azfile_create_dir(dir).await?;
+            let resp = self.azfile_create_dir(p).await?;
```

Comment:
This change seems wrong. 

Given this case: `a/b/c`:

- `create_dir("a/b/")` failed since `a/` doesn't exist.
- `create_dir("a/")` succeed.
- But write `a/b/c` still failed since `a/b/` doesn't exist.

Reply:
fix

I'm not sure I got it right in #3646 , My English is not good

Reply:
Sorry for not make it clear in this issue.

The `check` means sending `azfile_get_path_properties` call before calling `azfile_create_dir` to avoid extra API calls.

Take `a/b/c` as an example:

- Check if `a/b/` exist -> No
- Check if `a/` exist -> No
- Create `a/`
- Create `a/b/`

In the best case: if `a/b/` exists, we can avoid two APIs of create_dir.

Reply:
I think I understand. Thank you for your patience

I will close the pr